pkg/controller/orchestrator/orchestrator_reconcile.go

```diff
@@ -0,0 +1,478 @@
+	if timeSinceReady < healtyMoreThanMinutes {
+		log.Info("cluster not ready for acknowledge", "timeSinceReady", timeSinceReady, "trashold", healtyMoreThanMinutes)
+		return toAck
+	}
+
+	for _, recovery := range recoveries {
+		if !recovery.Acknowledged {
+			// skip if it's a new recovery, recovery should be older then <healtyMoreThanMinutes> minutes
+			startTime, err := time.Parse(time.RFC3339, recovery.RecoveryStartTimestamp)
+			if err != nil {
+				log.Error(err, "time parse error",
+					"recovery", recovery,
+				)
+				continue
+			}
+			if time.Since(startTime).Minutes() < healtyMoreThanMinutes {
+				// skip this recovery
+				log.Error(nil, "tries to recover to sson", "recovery", recovery)
+				continue
+			}
+
+			toAck = append(toAck, recovery)
+		}
+	}
+	return toAck
+}
+
+func (ou *orcUpdater) acknowledgeRecoveries(toAck []orc.TopologyRecovery) error {
+	comment := fmt.Sprintf("Statefulset '%s' is healty for more then %d minutes",
+		ou.cluster.GetNameForResource(api.StatefulSet), healtyMoreThanMinutes,
+	)
+
+	// acknowledge recoveries
+	for _, recovery := range toAck {
+		if err := ou.orcClient.AckRecovery(recovery.Id, comment); err != nil {
+			return err
+		}
+		ou.recorder.Event(ou.cluster, eventNormal, "RecoveryAcked",
+			fmt.Sprintf("Recovery with id %d was acked.", recovery.Id))
+	}
+
+	return nil
+}
+
+func (ou *orcUpdater) updateStatusForRecoveries(recoveries []orc.TopologyRecovery) {
+	var unack []orc.TopologyRecovery
+	for _, recovery := range recoveries {
+		if !recovery.Acknowledged {
+			unack = append(unack, recovery)
+		}
+	}
+
+	if len(unack) > 0 {
+		msg := getRecoveryTextMsg(unack)
+		ou.cluster.UpdateStatusCondition(api.ClusterConditionFailoverAck,
+			core.ConditionTrue, "PendingFailoverAckExists", msg)
+	} else {
+		ou.cluster.UpdateStatusCondition(api.ClusterConditionFailoverAck,
+			core.ConditionFalse, "NoPendingFailoverAckExists", "no pending ack")
+	}
+}
+
+// nolint: unparam
+func condIndexCluster(cluster *wrapcluster.MysqlCluster, condType api.ClusterConditionType) (int, bool) {
+	for i, cond := range cluster.Status.Conditions {
+		if cond.Type == condType {
+			return i, true
+		}
+	}
+
+	return 0, false
+}
+
+// updateNodeCondition is a helper function that updates condition for a specific node
+func (ou *orcUpdater) updateNodeCondition(host string, cType api.NodeConditionType, status core.ConditionStatus) {
+	ou.cluster.UpdateNodeConditionStatus(host, cType, status)
+}
+
+// removeNodeConditionNotInOrc marks nodes not in orc with unknown condition
+// TODO: this function should remove completly from cluster.Status.Nodes nodes
+// that are no longer in orchestrator and in k8s
+func (ou *orcUpdater) removeNodeConditionNotInOrc(insts InstancesSet) {
+	for _, ns := range ou.cluster.Status.Nodes {
+		node := insts.GetInstance(ns.Name)
+		if node == nil {
+			// node is NOT updated so will be marked as unknown all conditions types
```

Comment:
`node is NOT updated so will be marked as unknown all conditions types` ->  `node is NOT updated so all conditions will be marked as unknown`